### ISSUE TITLE
SIGSEGV after passing all tests

### ISSUE THREAD

**Issue by bwiklund** [cypress-io/cypress#7048]
I can't comment on the original ticket because it's locked, but this bug https://github.com/cypress-io/cypress/issues/6458 still exists on 4.4.0 as of this evening

![image](https://user-images.githubusercontent.com/3586801/79524736-1e0c9b80-8016-11ea-8cd8-4ea0ec5b3c08.png)

**Comment by jennifer-shehane** [comment]
@bwiklund Is your output also printing `Error: This socket has been ended by the other party`? If not, it may be another issue. Many things could cause a SIGSEGV error. 

Could you run Cypress in [debug mode](https://on.cypress.io/troubleshooting#Print-DEBUG-logs) mode and print the entire set of logs here?

**Comment by jamesdonoh** [comment]
Hi @jennifer-shehane my colleague reported https://github.com/cypress-io/cypress/issues/7073 which was closed in favour of this - can't speak for @bwiklund but with `DEBUG=cypress:*` the final part of my log is:
```
2020-04-20T16:29:09.804Z cypress:server:run attempting to close the browser
2020-04-20T16:29:09.804Z cypress:server:browsers killing browser process
2020-04-20T16:29:09.804Z cypress:server:electron-app all BrowserWindows closed, not exiting
2020-04-20T16:29:09.804Z cypress:server:browsers:electron closed event fired
2020-04-20T16:29:09.804Z cypress:server:windows removing all electron extensions []
2020-04-20T16:29:09.804Z cypress:server:preprocessor removeFile /Users/donohj03/repos/simorgh/cypress/integration/specialFeatures/cookieBanner/testsForAMPOnly.js
2020-04-20T16:29:09.804Z cypress:server:preprocessor base emitter plugin close event
2020-04-20T16:29:09.804Z cypress:server:preprocessor base emitter native close event
2020-04-20T16:29:09.804Z cypress:browserify close: /Users/donohj03/repos/simorgh/cypress/integration/specialFeatures/cookieBanner/testsForAMPOnly.js
2020-04-20T16:29:09.804Z cypress:browserify close: /Users/donohj03/repos/simorgh/cypress/integration/specialFeatures/cookieBanner/testsForAMPOnly.js
2020-04-20T16:29:09.804Z cypress:server:browsers browser process killed
2020-04-20T16:29:09.806Z cypress:server:run spec results { stats: { suites: 6, tests: 24, passes: 24, pending: 0, skipped: 0, failures: 0, wallClockStartedAt: 2020-04-20T16:28:55.805Z, wallClockEndedAt: 2020-04-20T16:29:09.766Z, wallClockDuration: 13961 }, reporter: 'spec', reporterStats: { suites: 6, tests: 24, passes: 24, pending: 0, failures: 0, start: 2020-04-20T16:28:55.816Z, end: 2020-04-20T16:29:09.799Z, duration: 13983 }, hooks: [ { hookId: 'h1', hookName: 'before each', title: [Array], body: 'function () {\n' + '                    visitPage(pageType, path);\n' + '                }' }, { hookId: 'h2', hookName: 'before each', title: [Array], body: 'function () {\n' + '                    visitPage(pageType, path);\n' + '                }' }, { hookId: 'h3', hookName: 'before each', title: [Array], body: 'function () {\n' + '                    visitPage(pageType, path);\n' + '                }' }, { hookId: 'h4', hookName: 'before each', title: [Array], body: 'function () {\n' + '
visitPage(pageType, path);\n' + '                }' }, { hookId: 'h5', hookName: 'before each', title: [Array], body: 'function () {\n' + '                    visitPage(pageType, path);\n' + '                }' }, { hookId: 'h6', hookName: 'before each', title: [Array], body: 'function () {\n' + '                    visitPage(pageType, path);\n' + '                }' } ], tests: [ { testId: 'r3', title: [Array], state: 'passed', body: 'function () {\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                    getPrivacyBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                    getCookieBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:55.816Z, wallClockDuration: 629, videoTimestamp: null }, { testId: 'r4', title: [Array], state: 'passed', body: 'function () {\n' + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:56.452Z, wallClockDuration: 395, videoTimestamp: null }, { testId: 'r5', title: [Array], state: 'passed', body: 'function () {\n' + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    getCookieBannerAccept(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:56.853Z, wallClockDuration: 507, videoTimestamp: null }, { testId: 'r6', title: [Array], state: 'passed', body: 'function () {\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    getCookieBannerReject(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:57.366Z, wallClockDuration: 1490, videoTimestamp: null }, { testId: 'r8', title: [Array], state: 'passed', body: 'function () {\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                    getPrivacyBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                    getCookieBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:58.864Z, wallClockDuration: 357, videoTimestamp: null }, { testId: 'r9', title: [Array], state: 'passed', body: 'function () {\n' + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:59.227Z, wallClockDuration: 302, videoTimestamp: null }, { testId: 'r10', title: [Array], state: 'passed', body: 'function () {\n' + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    getCookieBannerAccept(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:59.536Z, wallClockDuration: 368, videoTimestamp: null }, { testId: 'r11', title: [Array], state: 'passed', body: 'function () {\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                    getPrivacyBannerAccept(service, variant).click();\n' + '                    getCookieBannerReject(service, variant).click();\n' + '                    visitPage(pageType, path);\n' + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:28:59.910Z, wallClockDuration: 574, videoTimestamp: null }, { testId: 'r13', title: [Array], state: 'passed', body: 'function () {\n' + "                    getPrivacyBanner(service, variant).should('be.visible');\n" + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + '                    getPrivacyBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                    getCookieBannerAccept(service, variant).click();\n' + "                    getCookieBanner(service, variant).should('not.be.visible');\n" + "                    getPrivacyBanner(service, variant).should('not.be.visible');\n" + '                }', stack: null, error: null, timings: [Object], failedFromHookId: null, wallClockStartedAt: 2020-04-20T16:29:00.490Z, wallClockDuration: 948, videoTimestamp: null }, { testId: 'r14', titl────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  specialFeatures/cookieBanner/testsForCanonicalOnly.js                         (14 of 14)
  cypress:cli child event fired { event: 'exit', code: null, signal: 'SIGSEGV' } +2m
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: darwin (18.7.0)
Cypress Version: 4.4.0
```
I can share the whole file if you let me know the best way to do this. Thanks!

**Comment by jennifer-shehane** [comment]
@jamesdonoh Yeah, your SIGSEGV doesn't look to be logged after all the tests are run, but in between one spec before starting a new spec. The log of course gives no helpful information :(

**Comment by hindsc52** [comment]
> @jennifer-shehane is there anything else we can do to help diagnose the cause of this please? The issue we reported above was closed as a dupe of this one, and since migrating to 4.4.0 70-80% of our test runs now crash in this way.

Still seeing these crashed on 4/5 of our test runs. We upgraded to cypress 4.4.0 to move away from a security vulnerability in an older version and are reluctant to rollback re-introducing that security vuln.

Any updates would be appreciated.

**Comment by marinakuranda** [comment]
Just adding another report that the SIGSEGV is happening, with no useful log at all.

**Comment by apascal** [comment]
Hi,

We have the same issue here, reproduced with a minimal single test:

```js
describe('My First Test', () => {
  it('Visits the home', () => {
    cy.visit('/');
  });
});
```

`cypress.json` file is completely empty:
```json
{
}
```

Ran either in (gitlab-)ci with `cypress/included` docker image, or directly from the project root with this command on Mac OS Catalina:
```sh
CYPRESS_BASE_URL=http://localhost:3000 npx run cypress
```

Executed cypress version was `4.4.1`, but we've encountered this issue since at least `4.1.0`.

The error with DEBUG log activated is similar to what was previously indicated:
```sh
cypress:cli child event fired { event: 'exit', code: null, signal: 'SIGSEGV' } +13s
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV
```

Also note that we sometimes got the `SIGBUS` error instead: 
```sh
cypress:cli child event fired { event: 'exit', code: null, signal: 'SIGBUS' } +13s
The Test Runner unexpectedly exited via a exit event with signal SIGBUS
```

I then ran the following command:
```sh
DEBUG=* CYPRESS_BASE_URL=https://www.google.com npx cypress run
```

10 times out of 10, the result was a success by targeting the `google.com` home page rather than our own app.

Our web app renders much more slowly than the `google.com` home page, with quite a lot of javascript executed on the client side (8-10 seconds before first interactivity).

Hope this helps narrow down the problem.

Of course we are working on making our app faster (but don't have control over all scripts, and need to be able to run cypress while the app is in `development` mode, which is slower).

Is there a workaround to increase the limit (time / memory), for resource heavy web apps?
It used to work 100% well with more tests in cypress `3`, so maybe performance or tolerance was impacted with version `4`?

**Comment by abzainuddin** [comment]
Same problem seems to happen for me, in `4.5.0`:

```
 ================================================================================
  (Run Finished)
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV
Please search Cypress documentation for possible solutions:
https://on.cypress.io
Check if there is a GitHub issue describing this crash:
https://github.com/cypress-io/cypress/issues
Consider opening a new issue.
----------
Platform: linux (Debian - 10.3)
Cypress Version: 4.5.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by edsadr** [comment]
yeah.... same problem in our CI:

```
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:
https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

Platform: linux (Ubuntu Linux - 18.04)    
Cypress Version: 4.5.0
```

**Comment by mwren-mshanken** [comment]
Platform: linux (Debian - 8.10)
--
612 | Cypress Version: 4.5.0
613 | at /codebuild/output/src145582674/src/node_modules/cypress/lib/errors.js:326:15
614 | at tryCatcher (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/util.js:16:23)
615 | at Promise._settlePromiseFromHandler (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:547:31)
616 | at Promise._settlePromise (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:604:18)
617 | at Promise._settlePromise0 (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:649:10)
618 | at Promise._settlePromises (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:729:18)
619 | at Promise._fulfill (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:673:18)
620 | at Promise._resolveCallback (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:466:57)
621 | at Promise._settlePromiseFromHandler (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:559:17)
622 | at Promise._settlePromise (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:604:18)
623 | at Promise._settlePromise0 (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:649:10)
624 | at Promise._settlePromises (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:729:18)
625 | at Promise._fulfill (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:673:18)
626 | at Promise._resolveCallback (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:466:57)
627 | at Promise._settlePromiseFromHandler (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:559:17)
628 | at Promise._settlePromise (/codebuild/output/src145582674/src/node_modules/cypress/node_modules/bluebird/js/release/promise.js:604:18) known: true }
629 | ·[?25h

**Comment by jboler** [comment]
Same here using v4.5.0. Our specs pass locally but on Travis-CI fail after 2 minutes of the initial specs passing with:

    The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

After restarting the Travis build about 3 times it will complete successfully. 

Cypress 4.3 passes every time no problem. 

We are using cypress with TypeScript and no plugins. We are using fixtures only with quite a heavy API.

cypress.json:
```
{
  "supportFile": "cypress/support/index.ts",
  "videoUploadOnPasses": false,
  "baseUrl": "http://localhost:3030/_",
  "env": {
    "apiUrl": "http://localhost:3000/api/v2/"
  }
}
```

On Travis we're using a VM with 7.5Gb memory, no docker.

Travis config:
```
dist: xenial
language: node_js
node_js: 10

addons:
  chrome: stable

cache:
  yarn: true
  directories:
    - node_modules

env:
  - NODE_OPTIONS="--max_old_space_size=6144" # ensure enough memory for the tests

before_install:
  - curl -o- -L https://yarnpkg.com/install.sh | bash
  - export PATH=$HOME/.yarn/bin:$PATH

before_script:
  - yarn start --silent &

script:
  - set -e # Exit immediately if a subsequent command fails
  - yarn install --non-interactive --production=false --pure-lockfile
  - cypress install && yarn cypress run --config video=false
  - kill $(jobs -p) || true
```

**Comment by jboler** [comment]
With Cypress 4.6 the tests all complete but the Cypress process locks up logging the final output:

![image](https://user-images.githubusercontent.com/38783/82598622-949a3d00-9b70-11ea-9d6d-a798dfc19436.png)

**Comment by jboler** [comment]
Still failing with v4.8

**Comment by jboler** [comment]
So we did these 3 things and our Travis Cypress runs no longer hang. Not sure which one fixed it but I suspect it's the file watching because it's a very large codebase.

- Upgraded to Cypress 5.1
- Removed `@cypress/webpack-preprocessor` from package.json
- Disabled file events watchers for both cypress and our React app dev server with this:

.travis.yml - Added ` --config watchForFileChanges=false`
```yaml
  - cypress install && yarn cypress run --config watchForFileChanges=false
```

craco.config.js - added this:
```js
  ...(process.env.CI && {
    devServer: {
      watchOptions: {
        ignored: '/**',
      },
    },
  }),
```

**Comment by markodelgadillo** [comment]
Hi all... looks like I'm joining the party with this crash:

```
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: darwin (19.6.0)
Cypress Version: 5.1.0
```

I saved my changes which triggered the test to run again as I still had the test browser open from the previous test. This happened to me twice in a span of 5 minutes :(

**Comment by lexsmil** [comment]
Cypress:      5.4.0
Browser:      Chrome 86
OS:              CentOS 7.8.2003

`    ✔  All specs passed!                        00:04       20       20        -        -        -  

Error: The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: linux (CentOS - 7.8.2003)
Cypress Version: 5.4.0
    at /data/cypress/node_modules/cypress/lib/errors.js:346:15
    at tryCatcher (/data/cypress/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/data/cypress/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/data/cypress/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/data/cypress/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/data/cypress/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/data/cypress/node_modules/bluebird/js/release/promise.js:673:18)
    at Promise._resolveCallback (/data/cypress/node_modules/bluebird/js/release/promise.js:466:57)
    at Promise._settlePromiseFromHandler (/data/cypress/node_modules/bluebird/js/release/promise.js:559:17)
    at Promise._settlePromise (/data/cypress/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/data/cypress/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/data/cypress/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/data/cypress/node_modules/bluebird/js/release/promise.js:673:18)
    at Promise._resolveCallback (/data/cypress/node_modules/bluebird/js/release/promise.js:466:57)
    at Promise._settlePromiseFromHandler (/data/cypress/node_modules/bluebird/js/release/promise.js:559:17)
    at Promise._settlePromise (/data/cypress/node_modules/bluebird/js/release/promise.js:604:18) {
  known: true
}`

Tests are run via API and work perfectly on Cypress 5.3.0

**Comment by yunj** [comment]
Hi @jennifer-shehane I have encountered the same issue after upgrading from 5.3.0 to the latest 5.4.0 even with all tests passing. I use the docker image `cypress/browsers:node12.14.1-chrome85-ff81`.
```
2020-10-22T19:41:52.478Z cypress:server:cypress about to exit with code 0
2020-10-22T19:41:52.478Z cypress:rewriter:threads intercepted process.exit called, closing worker threads
2020-10-22T19:41:52.485Z cypress:rewriter:threads worker exited { exitCode: 0, worker: { isBusy: false, id: 4, freeWorkers: 4 } }
2020-10-22T19:41:52.498Z cypress:rewriter:threads worker exited { exitCode: 0, worker: { isBusy: false, id: 2, freeWorkers: 3 } }
2020-10-22T19:41:52.498Z cypress:rewriter:threads worker exited { exitCode: 0, worker: { isBusy: false, id: 3, freeWorkers: 2 } }
2020-10-22T19:41:52.518Z cypress:rewriter:threads worker exited { exitCode: 0, worker: { isBusy: false, id: 1, freeWorkers: 1 } }
2020-10-22T19:41:52.555Z cypress:https-proxy received error on client browserSocket { err: Error: write EPIPE at afterWriteDispatched (internal/stream_base_commons.js:154:25) at writeGeneric (internal/stream_base_commons.js:145:3) at Socket._writeGeneric (net.js:784:11) at Socket._write (net.js:796:8) at doWrite (_stream_writable.js:403:12) at writeOrBuffer (_stream_writable.js:387:5) at Socket.Writable.write (_stream_writable.js:318:11) at Socket.ondata (_stream_readable.js:695:22) at Socket.emit (events.js:310:20) at addChunk (_stream_readable.js:286:12) at readableAddChunk (_stream_readable.js:268:9) at Socket.Readable.push (_stream_readable.js:209:10) at TCP.onStreamRead (internal/stream_base_commons.js:186:23) { errno: 'EPIPE', code: 'EPIPE', syscall: 'write' }, url: 'example.com:443' }
2020-10-22T19:41:52.619Z cypress:rewriter:threads all workers terminated, exiting for real
2020-10-22T19:41:52.753Z cypress:cli child event fired { event: 'exit', code: null, signal: 'SIGSEGV' }
2020-10-22T19:41:52.760Z cypress:cli Stopping Xvfb
2020-10-22T19:41:52.767Z cypress:cli child event fired { event: 'close', code: null, signal: 'SIGSEGV' }
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: linux (Debian - 10.2)

Cypress Version: 5.4.0
```

**Comment by AntonioGargaro** [comment]
Commenting here as my post was closed as a duplicate, the same issue is occuring in our CI and local environments to error unexpectedly after all tests have passed successfully. We are using `Cypress 5.4.0` running in headless mode for E2E testing.

This is screenshot of local run failing with debug,
![Screenshot 2020-10-30 at 13 24 05](https://user-images.githubusercontent.com/38767071/97710087-307e8c00-1ab3-11eb-9dee-109aa340111d.png)

**Comment by cupidline** [comment]
![image](https://user-images.githubusercontent.com/16457311/97958941-dd714580-1de0-11eb-939d-55d65bb4676e.png)

Also crashes even though all spec is passed in the bitbucket pipeline. I have no problem with a crash, but this crash makes my test run failed even though all passed.

At my local also crash like this.

**Comment by mysticdevx** [comment]
Cypress version 5.6.0 in MacOS, when running through terminal, after all the tests cypress crashes and I am getting the same error: `The Test Runner unexpectedly exited via a exit event with signal SIGSEGV`

**Comment by jennifer-shehane** [comment]
@mysticdevx Are you able to provide a reproducible example? Even one you can email to support@cypress.io would be helpful. We'd really like to track down this blocking issue and this is an interesting situation you've mentioned.

**Comment by mysticdevx** [comment]
> @mysticdevx Are you able to provide a reproducible example? Even one you can email to [support@cypress.io](mailto:support@cypress.io) would be helpful. We'd really like to track down this blocking issue and this is an interesting situation you've mentioned.

[https://github.com/mysticdevx/SigsegvReproduce](SigsegvReproduceRepo)
pushed a very basic project having one test and only `experimentalSourceRewriting` property in Cypress.json along with very brief reproduce steps. Mailed too.

**Comment by jennifer-shehane** [comment]
@mysticdevx Thanks for providing this example. This led one of our engineers to open an issue specific to the `experimentalSourceRewriting` flag bug. 

There is a situation where Cypress will occasionally exit with SIGABRT, SIGTERM, or SIGSEGV and a non-zero exit code when `experimentalSourceRewriting` is set to true. Please follow this issue for updates if your SIG error disappears when setting `experimentalSourceRewriting: false`. https://github.com/cypress-io/cypress/issues/9208

We don't believe this is the issue for everyone in this thread however, since this issue was opened before `experimentalSourceRewriting` was released, so we will leave this issue open to track the cause of any other SIGSEGV errors.

**Comment by poponuts** [comment]
Just to add to this conversation, I had a similar issue running it on **Bitbucket pipeline**:
<img width="972" alt="Screen Shot 2020-12-21 at 9 55 27 pm" src="https://user-images.githubusercontent.com/36245304/102769911-69621f80-43d7-11eb-842b-9f6e53b941e0.png">
My `cypress.json` file contains **"experimentalSourceRewriting": true** (will change tomorrow if it works) and my plugins include percy.

**Comment by csymeonides-mf** [comment]
Just wanted to add that this problem appeared for me only when I migrated from `cy.server/cy.route` to `cy.intercept`. 

Setting `"experimentalSourceRewriting": false` has not helped.

Also, not sure if relevant, but I am using `"experimentalRunEvents": true` so that I can delete videos for passing tests [as suggested here](https://github.com/cypress-io/cypress/issues/2522#issuecomment-749316813).

Error log:
```
<--- Last few GCs --->
[299:0x116100000000]   755698 ms: Scavenge 2488.5 (2503.5) -> 2480.7 (2503.5) MB, 0.2 / 0.0 ms  (average mu = 0.941, current mu = 0.940) allocation failure 
[299:0x116100000000]   755714 ms: Scavenge 2488.5 (2503.5) -> 2480.7 (2503.5) MB, 0.1 / 0.0 ms  (average mu = 0.941, current mu = 0.940) allocation failure 
[299:0x116100000000]   755730 ms: Scavenge 2488.5 (2503.5) -> 2480.7 (2503.5) MB, 0.1 / 0.0 ms  (average mu = 0.941, current mu = 0.940) allocation failure 
<--- JS stacktrace --->
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV
Please search Cypress documentation for possible solutions:
https://on.cypress.io
Check if there is a GitHub issue describing this crash:
https://github.com/cypress-io/cypress/issues
Consider opening a new issue.
----------
Platform: linux (Debian - 10.6)
Cypress Version: 6.2.1
```

**Comment by csymeonides-mf** [comment]
Interesting: I noticed that, in the same test where this error occurs, if I revert back to using `cy.server/cy.route` I get the below warning (though the test still passes fine):

```
(node:568) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 preprocessor:close listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
```

**Comment by r0nunes** [comment]
> **SOLUTION**: Cypress version 5.6.0 in MacOS, when running through terminal, after all the tests complete, cypress crashes and I am getting the same error: `The Test Runner unexpectedly exited via a exit event with signal SIGSEGV`
> 
> I found the reason. the entry `"experimentalSourceRewriting": true` in the Cypress.json file causes the problem. when removed everything works like a charm. No error on terminal. Hope it works for you all

Thank you! works for me :)

**Comment by ngandymov** [comment]
I also have a problem with the CircleCi, if I run the test in the runner, it passes without an error, but as soon as I run it in the CircleCi, I get an error after the test and the test does not close runner/browser and freezes - the CircleCi gives an error with an inactivity timeout - what could it do be?

Test step:
1. Open site
2. Click to export (After clicking, the file will be downloaded)

If I comment out the step with download the file, then the test will pass successfully and close

Cypress: 6.6.0
CircleCi Orbs: cypress: cypress-io/cypress@1.27.0
 executor: cypress/browsers-chrome78-ff70
 browser: chrome

**CI output**
`
cypress:launcher chrome stderr: [0310/195328.586503:ERROR:nacl_helper_linux.cc(311)] NaCl helper process running without a sandbox!
Most likely you need to configure your SUID sandbox correctly +964ms
  cypress:launcher chrome exited: { code: null, signal: 'SIGSEGV' } +2ms
  cypress:server:preprocessor removeFile /root/project/packages/e2e/cypress/integration/smoke/testBug/checkExportOnAllEstimates.spec.js +36s
  cypress:server:preprocessor base emitter plugin close event +0ms
  cypress:server:preprocessor base emitter native close event +0ms
  cypress:server:preprocessor base emitter native close event +0ms
  cypress:webpack close /root/project/packages/e2e/cypress/integration/smoke/testBug/checkExportOnAllEstimates.spec.js +28s
  cypress:proxy:http:request-middleware request aborted +182ms
  cypress:server:request aborting { requestId: 'request91' } +97ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56946 } +563ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56924 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56746 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56752 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56866 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56916 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56956 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56868 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56870 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56720 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56744 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56542 } +1ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56826 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56840 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56750 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56566 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56710 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56522 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56534 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56856 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56914 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56748 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56872 } +0ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56976 } +1ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 55638 } +0ms
  cypress:https-proxy received error on client browserSocket { err: Error: write EPIPE     at afterWriteDispatched (internal/stream_base_commons.js:154:25)     at writeGeneric (internal/stream_base_commons.js:145:3)     at Socket._writeGeneric (net.js:786:11)     at Socket._write (net.js:798:8)     at doWrite (_stream_writable.js:403:12)     at writeOrBuffer (_stream_writable.js:387:5)     at Socket.Writable.write (_stream_writable.js:318:11)     at Socket.ondata (_stream_readable.js:716:22)     at Socket.emit (events.js:315:20)     at addChunk (_stream_readable.js:295:12)     at readableAddChunk (_stream_readable.js:271:9)     at Socket.Readable.push (_stream_readable.js:212:10)     at TCP.onStreamRead (internal/stream_base_commons.js:186:23) { errno: 'EPIPE', code: 'EPIPE', syscall: 'write' }, url: 'dev.shopmonkey.io:443' } +556ms
  cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 56944 } +3ms
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬──────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │         pids         │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼──────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │     'plugin'      │      1       │        '594'         │     0      │      0.24      │  361.8   │    329.2     │   377.77    │
  cypress:server:util:process_profiler │    1    │     'cypress'     │      1       │        '358'         │    2.41    │      0.76      │  230.3   │    202.11    │    230.3    │
  cypress:server:util:process_profiler │    2    │ 'electron-shared' │      4       │ '360, 513, 361, 552' │     0      │       0        │  170.47  │    152.97    │   170.47    │
  cypress:server:util:process_profiler │    3    │      'other'      │      3       │  '607, 1299, 1300'   │    3.3     │      0.66      │   3.52   │     3.46     │    3.52     │
  cypress:server:util:process_profiler │    4    │      'TOTAL'      │      9       │         '-'          │    5.71    │      2.15      │  766.09  │    956.97    │   1393.06   │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴──────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +10s
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬──────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │         pids         │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼──────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │     'plugin'      │      1       │        '594'         │     0      │      0.19      │  361.8   │    335.72    │   377.77    │
  cypress:server:util:process_profiler │    1    │     'cypress'     │      1       │        '358'         │    0.02    │      0.63      │  230.01  │    206.76    │    230.3    │
  cypress:server:util:process_profiler │    2    │ 'electron-shared' │      4       │ '360, 513, 361, 552' │     0      │       0        │  170.47  │    155.88    │   170.47    │
  cypress:server:util:process_profiler │    3    │      'other'      │      2       │     '1324, 1325'     │     0      │      0.55      │   3.43   │     3.46     │    3.52     │
  cypress:server:util:process_profiler │    4    │      'TOTAL'      │      8       │         '-'          │    0.02    │      1.8       │  765.71  │    925.1     │   1393.06   │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴──────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +10s
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬──────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │         pids         │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼──────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │     'plugin'      │      1       │        '594'         │     0      │      0.16      │  361.8   │    340.07    │   377.77    │
  cypress:server:util:process_profiler │    1    │     'cypress'     │      1       │        '358'         │    0.02    │      0.55      │  230.01  │    210.08    │    230.3    │
  cypress:server:util:process_profiler │    2    │ 'electron-shared' │      4       │ '360, 513, 361, 552' │     0      │       0        │  170.47  │    157.97    │   170.47    │
  cypress:server:util:process_profiler │    3    │      'other'      │      2       │     '1345, 1346'     │     0      │      0.47      │   3.41   │     3.45     │    3.52     │
  cypress:server:util:process_profiler │    4    │      'TOTAL'      │      8       │         '-'          │    0.02    │      1.54      │  765.7   │    902.32    │   1393.06   │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴──────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +10s
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:`

**Comment by ingin** [comment]
Hi @jennifer-shehane ,
I am also experimenting this issue with Cypress 6.4.0. Unfortunately, i can't remove the flag `"experimentalSourceRewriting": true,` because my website do programmatic redirection like `document.location.href = <target_page_url>`. If i remove this flag, Cypress will not able to perform redirection any more :(
![Capture du 2021-03-21 22-17-38](https://user-images.githubusercontent.com/4551229/111921191-7d2f2780-8a93-11eb-9982-a25c1190760e.png)

**Comment by flotwig** [comment]
@gaelcadoret This is a known issue with `experimentalSourceRewriting: true`, see the canonical issue: https://github.com/cypress-io/cypress/issues/9208

It will be fixed in 7.0.0.

**Comment by fugigoose** [comment]
We ran into this issue on 8.7.0 suddenly after adding a handful of additional tests using standard features/commands. No changes to config or anything, nothing different in these tests than the others. Tried enable debug logging (set `DEBUG=cypress:*` on Linux), but either was unable to or it didn't actually output anything except SIGSEGV. At the recommendation of @fmazu I downgraded to 8.5.0 and it has appeared to fix the problem. However, we seemed to be getting our SIGSEGV when trying to run a single spec file (and at the very beginning of that run). Tests run fine in `cypress open`. 

I wonder if there are multiple underlying issue here, but since it's failing with no specific error it's tough to say. The issue was seen on both Ubuntu 18.04 (5.4.0-91-generic) and Amazon Linux 2 (5.4.156-94.273.amzn2int.x86_64).

**Comment by abdullahsohailyaqoob** [comment]
Please help with this error

**Platform:** linux-x64 (Debian - 10.5)
**Cypress Version:** 9.2.1
The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ls@1.0.0 ksaStage: `cypress run --config-file cypress.ksa.json --env 
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ls@1.0.0 ksaStage script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-01-18T11_05_29_340Z-debug.log


Exited with code exit status 1
CircleCI received exit code

**Comment by pankajg10** [comment]
Same issue occur on latest version also.
Platform: linux-x64 (Red Hat Linux - 7.9)
Cypress Version: 9.4.1

The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Please suggest the stable version where this issue doesn't occur or shall i revert to 8.5 as mentioned in above comments?

**Comment by rakesh-basavaraju** [comment]
Same issue with me. I dont have the flag a flag experimentalSourceRewriting in cypress.json

Platform: linux-x64 (Ubuntu - 20.04)
Cypress Version: 9.5.0

The Test Runner unexpectedly exited via a exit event with signal SIGILL

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------
All my testing is blocked from couple of days. Please suggest any work arounds

**Comment by abdullahsohailyaqoob** [comment]
@rakesh-basavaraju , if you revert back cypress version to 7.6.0, hopefully that issue will resolve
Just a work around

**Comment by mattvb91** [comment]
Same issue started happening today when test suite starts on first test on github actions:

```
Platform: linux-x64 (Debian - 10.9)
Cypress Version: 9.5.3
```

**Comment by nhantrantr** [comment]
Same issue for me today:
Platform: linux-x64 (Ubuntu - 18.04)
Cypress Version: 9.5.3

**Comment by mattvb91** [comment]
So I decided to try and do a major upgrade on the docker image that runs cypress and went to:

```
FROM cypress/browsers:node17.8.0-chrome99-ff97-slim

```

still the exact same issue:

```
Platform: linux-x64 (Debian - 10.11)
Cypress Version: 9.6.1
```

**Comment by LaurentGoderre** [comment]
For us on Circle CI the issue os intermittent but in Docker running on Mac we seemed to be able to reproduce every time

**Comment by cooleiwhistles** [comment]
I am running into this same issue -> 
```
> npx cypress open -C cypress.config.js

The Test Runner unexpectedly exited via a exit event with signal SIGSEGV

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: linux-x64 (Ubuntu - 20.04)
Cypress Version: 10.11.0
```

are there any known workarounds?

**Comment by mattvb91** [comment]
@cooleiwhistles only `"cypress": "9.7.0"` is stable, i'd recommend trying that

**Comment by ardasendur** [comment]
we encountered same issue with cypress version 12.11.0. are there any known workarounds?

**Comment by mattvb91** [comment]
Ive managed to get it stable on '12.3.0'. Make sure you lock your package.jsons to exact versions and in docker images install the exact version too.

Only upgrade in small steps and validate each version works instead of having auto incrementing versions.

**Comment by ardasendur** [comment]
> Ive managed to get it stable on '12.3.0'. Make sure you lock your package.jsons to exact versions and in docker images install the exact version too.
> 
> Only upgrade in small steps and validate each version works instead of having auto incrementing versions.

We changed the version but error still exists. The error became "child event fired { event: 'close', code: null, signal: 'SIGTRAP' } "

**Comment by mschile** [comment]
@ardasendur, @mattvb91, or anyone else, would you be able to recreate the SIGSEGV fault in a reproduction repository by forking [cypress-test-tiny](https://github.com/cypress-io/cypress-test-tiny) or by other means? We would love to determine the cause of the failure instead of being forced to be on a previous version of Cypress.

**Comment by ardasendur** [comment]
> @ardasendur, @mattvb91, or anyone else, would you be able to recreate the SIGSEGV fault in a reproduction repository by forking [cypress-test-tiny](https://github.com/cypress-io/cypress-test-tiny) or by other means? We would love to determine the cause of the failure instead of being forced to be on a previous version of Cypress.

we will try after a while I will inform you.

**Comment by AtofStryker** [comment]
@ardasendur any luck with being able to provide a reproduction repository?

**Comment by mschile** [comment]
Right now there doesn't seem to be enough information to reproduce the SIGSEGV fault on our end. We'll have to close this issue until we can reproduce it. This does not mean the issue is not happening - it just means that we do not have a path to move forward.

Please comment with a reproducible example concerning the original issue and we can reopen.
